File: Exam Portal/ExamServer/src/main/java/com/exam/Service/QuizService.java
package com.exam.Service;

import com.exam.Entity.exam.Quiz;

import java.util.Set;

public interface QuizService {

    public Quiz addQuiz(Quiz quiz);
    public Quiz updateQuiz(Quiz quiz);
    public Set<Quiz> getQuizes();
    public Quiz getQuiz(Long quizId);
    public void deleteQuiz(Long quizId);
}

File: ExamFront/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseurl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private _http:HttpClient) { }

  public quizzes(){
    return this._http.get(`${baseurl}/quiz/`);
  }
  public addQuiz(quiz:any){
    return this._http.post(`${baseurl}/quiz/`,quiz);
  }
  public deleteQuiz(qId: any){
    return this._http.delete(`${baseurl}/quiz/${qId}`);
  }

  //get the single quiz
  public getQuiz(qId: any){
    return this._http.get(`${baseurl}/quiz/${qId}`)
  }

  public updateQuiz(quiz:any){
    return this._http.put(`${baseurl}/quiz/`,quiz);
  }
}
